tests/ui/sized/unsized-binding.stderr

```diff
@@ -1,13 +1,51 @@
+error[E0746]: return type cannot have an unboxed trait object
+  --> $DIR/unsized-binding.rs:4:13
+   |
+LL | fn bar() -> dyn T {
+   |             ^^^^^ doesn't have a size known at compile-time
+   |
+   = help: if the returned value came from a borrowed argument that implements the trait, then you could return `&dyn Trait`
+help: return an `impl Trait` instead of a `dyn Trait`
+   |
+LL | fn bar() -> impl T {
+   |             ~~~~
+help: alternatively, box the return type to make a boxed trait object, and wrap all of the returned values in `Box::new`
+   |
+LL ~ fn bar() -> Box<dyn T> {
+LL ~     Box::new(panic!())
+   |
+
 error[E0277]: the size for values of type `str` cannot be known at compilation time
-  --> $DIR/unsized-binding.rs:2:9
+  --> $DIR/unsized-binding.rs:9:13
    |
 LL |     let x = *"";
-   |         ^ doesn't have a size known at compile-time
+   |             ^^^ doesn't have a size known at compile-time
    |
    = help: the trait `Sized` is not implemented for `str`
    = note: all local variables must have a statically known size
    = help: unsized locals are gated as an unstable feature
 
-error: aborting due to 1 previous error
+error[E0277]: the size for values of type `dyn T` cannot be known at compilation time
+  --> $DIR/unsized-binding.rs:12:28
+   |
+LL |     let Foo(a, b) = Foo(1, bar());
+   |                            ^^^^^ doesn't have a size known at compile-time
+   |
+   = help: the trait `Sized` is not implemented for `dyn T`
+   = note: all local variables must have a statically known size
+   = help: unsized locals are gated as an unstable feature
+
+error[E0277]: the size for values of type `dyn T` cannot be known at compilation time
+  --> $DIR/unsized-binding.rs:12:28
+   |
+LL |     let Foo(a, b) = Foo(1, bar());
+   |                            ^^^^^ doesn't have a size known at compile-time
+   |
+   = help: the trait `Sized` is not implemented for `dyn T`
+   = note: all function arguments must have a statically known size
+   = help: unsized fn params are gated as an unstable feature
```

Comment:
This two could be deduplicated if we reworded the last two note and help to have the same text.

Reply:
The last commit does this. In stable, these two errors will collapse into a single one.